x-pack/plugins/apm/public/context/breadcrumbs/use_breadcrumb.ts

```diff
@@ -21,22 +21,13 @@ export function useBreadcrumb(breadcrumb: Breadcrumb | Breadcrumb[]) {
 
   const matchedRoute = useRef(match?.route);
 
-  useEffect(() => {
-    if (matchedRoute.current && matchedRoute.current !== match?.route) {
-      api.unset(matchedRoute.current);
-    }
-
-    matchedRoute.current = match?.route;
+  if (matchedRoute.current && matchedRoute.current !== match?.route) {
+    api.unset(matchedRoute.current);
+  }
 
-    if (matchedRoute.current) {
-      api.set(matchedRoute.current, castArray(breadcrumb));
-    }
+  matchedRoute.current = match?.route;
 
-    return () => {
-      if (matchedRoute.current) {
```

Comment:
We probably still need to unset it though.

Reply:
I did some tests and I didn't see much difference when the callback was there. I can put it back though just in case